src/type.js

```diff
@@ -11,8 +11,8 @@ export type TypeChecker<A> = (a: A) => A
 export type Type<A, B> = {
   '@@type': A,
   '@@value': B,
-  inspect(): String,
-  is(x: Data<Type<A, B>, any>): boolean
+  inspect: () => string,
+  is: (x: Data<Type<A, B>, any>) => boolean,
```

Comment:
We seem to be defining function signatures as `type Name = (a: A) => A` everywhere, but in the compound type definitions we were using the `name(a: A): A` syntax instead.

This makes it the same everywhere.